making .values return a numpy array

choldgraf opened the pull request:
this PR makes `table.values` return a numpy array (since it was being deprecated and isn't used in the textbook). This makes its behavior roughly equivalent to pandas dataframes. addresses #226 @SamLau95 @stefanv

choldgraf commented:
I could change this to call `table.to_df().values` instead, though figured I'd try it with pure numpy first.

SamLau95 commented on datascience/tables.py:
```diff
@@ -266,12 +266,16 @@ def column(self, index_or_label):
-        warnings.warn("values is deprecated; use column", FutureWarning)
-        return self.column(label)
+    @property
+    def values(self):
```
Think you could add a docstring + a test for this function? :)

SamLau95 commented:
cc'ing @papajohn . This LGTM after adding docstring + test. Oh, and could you also add a note in the CHANGELOG.md?

(Might also want to bump the version number if you need to `pip install` this right away.)

choldgraf commented:
Ah looks like this'll break a test: L493 of test_tables.py  (this tests the deprecated `values` function). Do you guys think it's OK to change this test?

choldgraf commented:
(and not a big rush on this)

papajohn commented on datascience/tables.py:
```diff
@@ -266,12 +266,16 @@ def column(self, index_or_label):
+            dtype = None
+        return np.array(self.rows, dtype=dtype)
 
+    # Deprecated
```
Is this supposed to be deprecated, or did the comment just move from the old values function?

choldgraf replied:
just moved from the old values function...I assume it was a section header?

papajohn replied:
Please remove the comment. `values` was deprecated, but `column_index` is not. Thanks.

papajohn commented:
I have no problem with replacing the old `.values()`. Documentation would be great, as @SamLau95  suggested.

choldgraf commented:
Sounds good - I'll add a docstring + test

choldgraf commented:
Also, looks like I'm an idiot and mistook a dictionary for a table 😆 ...I don't think this will break a test after all but I'll add a new one regardless

choldgraf commented:
Mmk, added a test for this as well as a docstring. Also removed the `# Deprecated` comment because I think it was attached only to the deprecated `values()` function after all

stefanv commented:
Looks good to me.

SamLau95 commented:
LGTM too! 🚢

choldgraf commented:
hooray!

On Tue, Jul 19, 2016 at 12:49 AM, Sam Lau <notifications@github.com> wrote:

> Merged #227 <https://github.com/data-8/datascience/pull/227>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/data-8/datascience/pull/227#event-727068439>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABwSHWYYCZywMEf8u-Z1lJpV1WxBVXI2ks5qXGWHgaJpZM4JPWTR>
> .
>
